src/hooks/useRequestProcessor.js

```diff
@@ -0,0 +1,66 @@
+import { useCallback } from 'react'
+import { useHeartbeat } from './useCourtContracts'
+import { useCourtClock } from '../providers/CourtClock'
+import { useRequestQueue } from '../providers/RequestQueue'
+import { useActivity } from '../components/Activity/ActivityProvider'
+
+import radspec from '../radspec'
+import allowedTermsBehind from '../allowedTermsBehind'
+
+export function useRequestProcessor() {
+  const { request } = useHeartbeat()
+  const { addActivity } = useActivity()
+  const { addRequests } = useRequestQueue()
+  const { neededTransitions } = useCourtClock()
+
+  const processRequests = useCallback(
+    requests => {
+      const processedRequests = []
+        .concat(requests)
+        .map(
+          ({
+            action,
+            description,
+            ensureConfirmation = false,
+            isTx = true,
+            name,
+            params,
+            ...request
+          }) => {
+            const intent = isTx
+              ? () => addActivity(action(), name, params)
+              : action
+
+            return {
+              name,
+              intent,
+              description:
+                description || radspec[name](...Object.values(params)),
+              ensureConfirmation,
+              isTx,
+              ...request,
+            }
+          }
+        )
+
+      const maxAllowedTermsBehind = processedRequests
+        .filter(req => req.isTx)
+        .reduce((acc, req) => Math.max(acc, allowedTermsBehind[req.name]), -1)
```

Comment:
The `allowedTermsBehind[req.name]` just feels a _little_ bit sketchy, since we're using the name (which may be mistyped or disconnected from the actual action).

Not sure if there's a good solution for this though.

Reply:
hmm maybe we could change `name` to be `type`? Because now that you mention it, the name is the actually the `court-action-type` that now we are changing it to be a `Symbol` as you suggested.

Reply:
Yes, let's do that!